### PR TITLE
fix: correct negative-order detection in is_continuous_axis

### PR DESCRIPTION
**Pull request by shriramThakare3**

### CODE REVIEW

File: keras/src/utils/python_utils.py [hunk 1]
```diff
@@ -6,21 +6,14 @@
 
 
 def is_continuous_axis(axis):
-    # Used to determine whether the dimensions in an axis are continuous
-    if isinstance(axis, int) or len(axis) == 1:
+    if len(axis) <= 1:
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/utils/python_utils.py]: +1 please re-add the `isinstance(axis, int)`.

File: keras/src/utils/python_utils.py [hunk 2]
```diff
@@ -6,21 +6,14 @@
 
 
 def is_continuous_axis(axis):
-    # Used to determine whether the dimensions in an axis are continuous
```
Comment (hertschuh) [inline keras/src/utils/python_utils.py:9]: Please re-add the comment that was there, it's not entirely obvious what this function does.

**Review by hertschuh — COMMENTED** [review]
Thank you for the improvement!

File: keras/src/utils/python_utils.py (hunk 2 above)
Comment (shriramThakare3) [reply keras/src/utils/python_utils.py:9]: Done — re-added the isinstance(axis, int) check and the comment.

File: keras/src/utils/python_utils.py [hunk 3]
```diff
@@ -7,20 +7,14 @@
 
 def is_continuous_axis(axis):
     # Used to determine whether the dimensions in an axis are continuous
-    if isinstance(axis, int) or len(axis) == 1:
+    if isinstance(axis,int) or len(axis) <= 1:
```
Comment (hertschuh) [inline keras/src/utils/python_utils.py]: You need the space between `axis,` and `int`. This is why the code format check is failing.

**Review by hertschuh — CHANGES_REQUESTED** [review]

File: keras/src/utils/python_utils.py (hunk 3 above)
Comment (shriramThakare3) [reply keras/src/utils/python_utils.py]: added space

**Comment by shriramThakare3** [comment]
### Summary

Fixes #22162

`PReLU` and `LeakyReLU` lacked sufficient input validation, allowing
broken layer configurations to be created silently and only failing
later during training or serialization.

AI tool used: Claude Sonnet 4.6

- [x] I have read the CONTRIBUTING document.
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.

### Details and comments

**`keras/src/layers/activations/prelu.py`**
Added a `None` check for `alpha_initializer` in `__init__`. Passing
`None` previously caused a silent broken state that deferred errors to
`build()` or training. A `ValueError` is now raised immediately with a
clear message pointing to the correct fix.

**`keras/src/layers/activations/leaky_relu.py`**
Added `math.isnan()` to the existing validation condition. `NaN` values
were previously accepted and would produce non-numeric gradients silently
during training. The check is folded into the existing `if` guard
alongside the existing `None` and negative value checks, and the error
message is updated to mention `NaN`.

Before:
```python
if negative_slope is None or negative_slope < 0:
```
After:
```python
if negative_slope is None or math.isnan(negative_slope) or negative_slope < 0:
```

Both fixes follow the fail-fast principle already used elsewhere in
Keras — invalid configurations are caught at layer instantiation rather
than deferred to the training phase.

**Comment by shriramThakare3** [comment]
@hertschuh removed the changes in prelu and leaky_relu

**Comment by shriramThakare3** [comment]
@hertschuh  
changes are  made , but in prelu there is "+" showing but not in my code
apologizes i dont have idea about that
